pkg/cfn/builder/nodegroup.go

```diff
@@ -114,14 +114,28 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 	}
 
 	if volumeSize := n.spec.VolumeSize; volumeSize != nil && *volumeSize > 0 {
-		launchTemplateData.BlockDeviceMappings = []gfn.AWSEC2LaunchTemplate_BlockDeviceMapping{{
-			DeviceName: gfn.NewString(*n.spec.VolumeName),
-			Ebs: &gfn.AWSEC2LaunchTemplate_Ebs{
-				VolumeSize: gfn.NewInteger(*volumeSize),
-				VolumeType: gfn.NewString(*n.spec.VolumeType),
-				Encrypted: gfn.NewBoolean(*n.spec.VolumeEncrypted),
-			},
-		}}
+		if api.IsSetAndNonEmptyString(n.spec.VolumeKmsKeyID) {
```

Comment:
You can do here something like:

```
launchTemplateData.BlockDeviceMappings = []gfn.AWSEC2LaunchTemplate_BlockDeviceMapping{{
				DeviceName: gfn.NewString(*n.spec.VolumeName),
				Ebs: &gfn.AWSEC2LaunchTemplate_Ebs{
					VolumeSize: gfn.NewInteger(*volumeSize),
					VolumeType: gfn.NewString(*n.spec.VolumeType),
					Encrypted: gfn.NewBoolean(*n.spec.VolumeEncrypted),
				},
			}}
if api.IsSetAndNonEmptyString(n.spec.VolumeKmsKeyID) {
  launchTemplateData.BlockDeviceMappings.Ebs.KmsKeyId = gfn.NewString(*n.spec.VolumeKmsKeyID)
}
```

Reply:
Ah cool, did not realise I could do it that way. Thanks!